packages/component/src/createLoadable.js

```diff
@@ -184,15 +187,8 @@ function createLoadable({ resolve = identity, render, onLoad }) {
         const { error, loading, result } = this.state
 
         if (options.suspense) {
-          const cachedResult = this.getCache()
-          if (!cachedResult) throw this.loadAsync()
-          return render({
-            loading: false,
-            fallback: null,
-            result: cachedResult,
-            options,
-            props: { ...props, ref: forwardedRef },
-          })
+          const cachedPromise = this.getCache()
+          if (!cachedPromise) throw this.loadAsync()
```

Comment:
That would happen if you have two components? What if this component would be rendered twice?
In both cases the second one would be  "cached" and probably would not work.

Probably the cache should hold information about the status of a cached item.

Reply:
If you have 2 components at the same tree level, they should have different Keys (like in arrays), having different Keys makes them different instances with their own internal cache

What are your thoughts on that?

Reply:
You might code split  absolutely everything, including a single `Button`, you can use 20 times on a single page.
So - it could be the same cache key, repeated 20 times.

Reply:
I've added some tests for multiple rendered components
Please check them out to see if I understood you correctly